src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java

```diff
@@ -126,41 +128,49 @@ private void parseModsGroup(Map<String, String> fields, List<Object> modsGroup,
-            } else if (groupElement instanceof LanguageDefinition) {
-                LanguageDefinition language = (LanguageDefinition) groupElement;
-                putIfValueNotNull(fields, FieldName.LANGUAGE, language.getLangValue());
-            } else if (groupElement instanceof LocationDefinition) {
-                LocationDefinition location = (LocationDefinition) groupElement;
-                parseLocationAndUrl(fields, location);
-            } else if (groupElement instanceof NameDefinition) {
-                NameDefinition name = (NameDefinition) groupElement;
-                handleAuthorsInNamePart(name, authors, fields);
-            } else if (groupElement instanceof OriginInfoDefinition) {
-                OriginInfoDefinition originInfo = (OriginInfoDefinition) groupElement;
-                parseOriginInfo(fields, originInfo);
-            } else if (groupElement instanceof RecordInfoDefinition) {
-                RecordInfoDefinition recordInfo = (RecordInfoDefinition) groupElement;
-                parseRecordInfo(fields, recordInfo);
-            } else if (groupElement instanceof NoteDefinition) {
-                NoteDefinition note = (NoteDefinition) groupElement;
-                notes.add(note.getValue());
-            } else if (groupElement instanceof RelatedItemDefinition) {
-                RelatedItemDefinition relatedItem = (RelatedItemDefinition) groupElement;
-                List<Object> relatedModsGroup = relatedItem.getModsGroup();
+
+            //Get the element. Only one of the elements should be not an empty optional. This element will be catched
+            //by the if clauses below.
+            Optional<AbstractDefinition> abstractDefinition = getElement(groupElement, AbstractDefinition.class);
+            Optional<GenreDefinition> genre = getElement(groupElement, GenreDefinition.class);
+            Optional<LanguageDefinition> languageDefinition = getElement(groupElement, LanguageDefinition.class);
+            Optional<LocationDefinition> location = getElement(groupElement, LocationDefinition.class);
+            Optional<NameDefinition> name = getElement(groupElement, NameDefinition.class);
+            Optional<OriginInfoDefinition> originInfo = getElement(groupElement, OriginInfoDefinition.class);
+            Optional<RecordInfoDefinition> recordInfo = getElement(groupElement, RecordInfoDefinition.class);
+            Optional<NoteDefinition> note = getElement(groupElement, NoteDefinition.class);
+            Optional<RelatedItemDefinition> relatedItem = getElement(groupElement, RelatedItemDefinition.class);
+            Optional<SubjectDefinition> subject = getElement(groupElement, SubjectDefinition.class);
+            Optional<IdentifierDefinition> identifierDefinition = getElement(groupElement, IdentifierDefinition.class);
+            Optional<TitleInfoDefinition> titleInfo = getElement(groupElement, TitleInfoDefinition.class);
+
+            if (abstractDefinition.isPresent()) {
```

Comment:
Like the idea with the Optionals :+1:  Keep on the good work!
You could even simplify that more: if you just use the `ifPresent `of Optional (accepts a lambda) you can omit those if/else if complety and the check of isPresent, because as the name already says it gets only executed if the value in the optional is present.